python/tskit/tables.py

```diff
@@ -335,16 +336,41 @@ def __setattr__(self, name, value):
 
     def __getitem__(self, index):
         """
-        Return the specified row of this table, decoding metadata if it is present.
-        Supports negative indexing, e.g. ``table[-5]``.
+        If passed an integer, return the specified row of this table, decoding metadata
+        if it is present. Supports negative indexing, e.g. ``table[-5]``.
+        If passed a slice, iterable or array return a new table containing the specified
+        rows. Similar to numpy fancy indexing, if the array or iterables contains
+        booleans then in acts as a mask, returning those rows for which are True.
+
+        :param index: the zero-index of a desired row, a slice of the desired rows, an
+            iterable or array of the desired row numbers, or a boolean array to use as
+            a mask.
+        """
+
+        if isinstance(index, numbers.Integral):
+            # Single row by integer
```

Comment:
Out of curiosity, could you do ``index = slice(index).indices(len(self))[0]``?

More obscure, I guess, but saves a bit of fiddly code?

Reply:
Unfortunately not:
```python
>>> slice(5).indices(10)
(0, 5, 1)
```